packages/api-client/src/collections.ts

```diff
@@ -13,16 +16,24 @@ const { invokeApi } = require('./cumulusApiClient');
  *                                     to cumulusApiClient.invokeApi
  * @returns {Promise<Object>}          - the response from the callback
  */
-const createCollection = async ({ prefix, collection, callback = invokeApi }) => callback({
-  prefix,
-  payload: {
-    httpMethod: 'POST',
-    resource: '/{proxy+}',
-    headers: { 'Content-Type': 'application/json' },
-    path: '/collections',
-    body: JSON.stringify(collection)
-  }
-});
+export const createCollection = async (params: {
+  prefix: string,
```

Comment:
Given how often we're using an interface with the shape 

{ 
   prefix: string,
   callback?: InvokeApiFunction,
   <other things>: 
} 

I'm wondering if moving some of that boilerplate/inheritance to types.ts might make more sense. 

It's probably YAGNI, but if we decided to change the requirements for the callback there'd be a lot of boilerplate to update versus a base interface.  

Not something to hold the PR up on, as this is still a direct improvement, but a consideration for a pattern/refactor discussion.

Reply:
This kind of relates to the discussion I had with @markdboyd earlier. There are cases where there is some repetition, but I'm worried that over-DRYing things will add too much additional complexity for the benefit we get from it. This feels like one of those situations. I guess it's subjective, but those two params don't feel like they have anything in common other than the fact that a bunch of these functions happen to use them.

Reply:
I mea.... they happen to use them because they're all just mapping <arguments> to a <payload> and then doing the same thing.   Perhaps there's refactoring we could do to address it too.   In either case that can be punted to an arch discussion.